### DISCUSSION TITLE
Can't get Extensions to work

### DISCUSSION THREAD

**Discussion by Michael-Gerlach** [ahrm/sioyek#959] [General]
Hi and thanks for the great application!

I wanted to install a few extensions to Sioyek, but somehow can't get it to work. 

I installed Sioyek on my Mac via brew 
`brew install --cask sioyek`

and the executable is located in '/Applications/sioyek.app'

I installed the python wrapper via 
`python -m pip install sioyek`

I added the configuration to my prefs.user file:
`new_command _embed_annotations python -m sioyek.embed_annotations "%{sioyek_path}" "%{local_database}" "%{shared_database}" "%{file_path}"`

Now the new command shows up in the Sioyek menu, but I nothing happens upon execution, for other extensions I get the same behavior.

I tried changing `python` to `python3` and `python3.8` and `"%{sioyek_path}" ` to `'/Applications/sioyek.app'`, but that did not change anything. 

If I run 

`python -m sioyek.embed_annotations "%{sioyek_path}" "%{file_path}" "%{command_text}"`

I get the error:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "%{file_path}", line 7, in <module>
    SHARED_DATABASE_PATH = clean_path(sys.argv[3])
                                      ~~~~~~~~^^^
IndexError: list index out of range

```

What am I doing wrong here? Is there anything else I can try?

**Comment by Michael-Gerlach** [comment]
I eventually figured out that I have to specify the path to the correct python version in the prefs_user.config file:

`new_command _embed_annotation /Users/username/opt/anaconda3/bin/python -m sioyek.embed_annotations "%{sioyek_path}" "%{local_database}" "%{shared_database}" "%{file_path}" `

Unfortunately, now some highlights are embedded incorrectly, just as described [here](https://github.com/ahrm/sioyek/issues/879).

However, the solution presented there does not work for me.

**Comment by MrJuna1d** [comment]
I am trying to install the add_text extension but I am facing a similar issue where the command is being recognised but nothing happens and I am using the specific path for the python version

Can you tell me one your folder placement, where did you save the .py file and where is your prefs_user.config file??

**Reply by MrJuna1d** [reply]
got it working, Apparently I had an architecture mismatch, I had installed arm version but sioyek supports x86 so had to delete and reinstall sioyek dependency.
